bindings/cs/MongoDB.Libmongocrypt/Binary.cs

```diff
@@ -77,14 +77,16 @@ public byte[] ToArray()
         /// </summary>
         public void WriteBytes(byte[] bytes)
         {
-            if (Length == bytes.Length)
+            // The length of the new bytes can be smaller than allocated memory 
+            // because sometimes the allocated memory contains reserved blocks for future usage
+            if (Length >= bytes.Length)
```

Comment:
I think we're trying to determine if it's safe to copy the bytes? The test makes more sense the other way around:

    if (bytes.Length <= Length)